### DISCUSSION TITLE
RTSP Proxy to HLS?

### DISCUSSION THREAD

**Discussion by LaurenceJJones** [bluenviron/mediamtx#501] [Questions]
I set up 4 RTSP proxy cams, whenever I try to connect via HLS I get this output:
2021/08/12 16:46:19 I [0/0] rtsp-simple-server v0.0.0
2021/08/12 16:46:19 I [0/0] [path proxied1] created
2021/08/12 16:46:19 I [0/0] [path proxied2] created
2021/08/12 16:46:19 I [0/0] [path proxied3] created
2021/08/12 16:46:19 I [0/0] [path proxied4] created
2021/08/12 16:46:19 I [0/0] [RTSP] UDP/RTP listener opened on :8000
2021/08/12 16:46:19 I [0/0] [RTSP] UDP/RTCP listener opened on :8001
2021/08/12 16:46:19 I [0/0] [RTSP] TCP listener opened on :8554
2021/08/12 16:46:19 I [0/0] [RTMP] listener opened on :1935
2021/08/12 16:46:19 I [0/0] [HLS] listener opened on :8888
2021/08/12 16:46:28 I [0/0] [HLS] [conn 172.17.0.1:44954] GET /proxied3/
2021/08/12 16:46:28 I [0/0] [HLS] [remuxer proxied3] created
2021/08/12 16:46:28 I [0/0] [path proxied3] [rtsp source] started
2021/08/12 16:46:28 I [0/0] [path proxied3] [rtsp source] ready
2021/08/12 16:46:28 I [0/0] [HLS] [remuxer proxied3] ERR: unable to find a video or audio track
2021/08/12 16:46:28 I [0/0] [HLS] [remuxer proxied3] destroyed
2021/08/12 16:46:38 I [0/0] [path proxied3] [rtsp source] stopped
2021/08/12 16:47:00 I [0/0] [HLS] [conn 172.17.0.1:44962] GET /proxied3/stream.m3u8
2021/08/12 16:47:00 I [0/0] [HLS] [remuxer proxied3] created
2021/08/12 16:47:00 I [0/0] [path proxied3] [rtsp source] started
2021/08/12 16:47:00 I [0/0] [path proxied3] [rtsp source] ready
2021/08/12 16:47:00 I [0/0] [HLS] [remuxer proxied3] ERR: unable to find a video or audio track
2021/08/12 16:47:00 I [0/0] [HLS] [remuxer proxied3] destroyed
2021/08/12 16:47:10 I [0/0] [path proxied3] [rtsp source] stopped

What am I doing wrong here?

EDIT: it seems my streams do not work, cheap DVR boxes like to append details about the stream here an example
rtsp://<ip>:554/user=&password=&channel=3&stream=0.sdp

**Comment by aler9** [comment]
Hello, the problem here is that HLS requires streams with tracks encoded either with H264 or AAC, and your stream probably uses other encodings. Anyway, the error message can be certainly improved to be clearer.

**Reply by aler9** [reply]
error message is clearer in v0.17.2

[a comment by an automated account is not shown]
